aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java

```diff
@@ -276,16 +276,20 @@ private SelectionSet wrapPagination(SelectionSet node) {
          */
         private static boolean isCustomType(@NonNull Field field) {
             Class<?> cls = getClassForField(field);
-            if (Model.class.isAssignableFrom(cls) || Enum.class.isAssignableFrom(cls)) {
+            if (cls.getClassLoader() == null || isDateType(cls)) {
```

Comment:
@jamesonwilliams in java any jdk preloaded classes like ```Integer```, ```String``` （anything inside **rt.jar**） will use bootstrap classloader which is a native one, so in java it will show as null, any application classes (custom type) will use a concrete application classloader.